RELEASE.md

```diff
@@ -41,58 +41,70 @@ with `make upgrade-npm-deps`
+  - Entries in the `CHANGELOG.md` are meant to be in this order:
+    * `[FEATURE]`
+    * `[ENHANCEMENT]`
+    * `[BUGFIX]`
+    * `[BREAKINGCHANGE]`
+  - Entries that map to a pull request should include a pull request number.
+  - As we have many libraries we publish, it's better if you also put a clear indication about what library is affected by
+    these changes.
+  - Consumers understand how to handle breaking changes either through the messaging in the changelog or through the linked pull requests.
+- Update the `package.json` files for all packages with the corresponding version:
+
+  ```bash
+  make bump-version
+  ```
+- Push the branch to github and create a pull request with the release branch as the base. This gives others the opportunity to chime in on the release,
+  in general, and on the addition to the changelog, in particular.
+  - It's also helpful to drop a link to the release PR in #perses-dev on Matrix to get extra visibility.
+- Address any necessary feedback.
+- Once the pull request is approved, merge the it into the release branch.
+
+### 2. Create release tag and validate release
+
+- Pull down the latest updates to the release branch on your local machine to ensure you have the updates from the previous step.
+- Tag the new release via the following commands:
+
+  ```bash
+  git checkout release/v<version_to_be_replaced>
+  make tag
+  git push origin v<version_to_be_replaced>
+  ```
 
-### 2. Draft the new release
+Signing a tag with a GPG key is appreciated, but in case you can't add a GPG key to your Github account using the
+following [procedure](https://docs.github.com/en/authentication/managing-commit-signature-verification), you can replace
+the `-s` flag by `-a` flag of the git tag command to only annotate the tag without signing. If you are using a newer Mac and hit an error like "gpg failed to sign the data fatal: failed to write commit object," see [this Stack Overflow question](https://stackoverflow.com/questions/39494631/gpg-failed-to-sign-the-data-fatal-failed-to-write-commit-object-git-2-10-0/40066889#40066889) for assistance.
 
-Tag the new release via the following commands:
+Once a tag is created, an automated release process for this tag is triggered via Github Actions. This automated process includes:
+- Publishing new versions of the UI libraries to npm.
+- Building new go binaries and docker images.
+- Publishing the docker images to Docker Hub.
+- Creating a new Github release that uses the changelog as the release notes and provides tarballs with the latest go binaries.
 
-```bash
-git checkout release/v<version_to_be_replaced>
-make tag
-git push origin v<version_to_be_replaced>
-```
+Please verify that the Github Actions complete successfully. Once they are completed, check that everything looks good in the new Github Release. If you realize we need to adjust something in the release notes, you can edit it directly in the Github UI.
```

Comment:
not big deal, sometimes we're camel-casing GitHub and sometimes we're all lowercasing it (github).